internal/services/cdn/cdn_frontdoor_route_resource.go

```diff
@@ -321,89 +385,128 @@ func resourceCdnFrontDoorRouteUpdate(d *pluginsdk.ResourceData, meta interface{}
 	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.FrontDoorRouteID(d.Id())
+	id, err := parse.FrontDoorRouteIDInsensitively(d.Id())
 	if err != nil {
 		return err
 	}
 
-	// NOTE: I now need to lock this resource when updating because the
-	// cdn_frontdoor_route_unlink_default_domain resources may also be
-	// trying to update it as well...
-	locks.ByName(id.RouteName, cdnFrontDoorRouteResourceName)
-	defer locks.UnlockByName(id.RouteName, cdnFrontDoorRouteResourceName)
-
 	existing, err := client.Get(ctx, id.ResourceGroup, id.ProfileName, id.AfdEndpointName, id.RouteName)
 	if err != nil {
 		return fmt.Errorf("retrieving existing %s: %+v", *id, err)
 	}
+
 	if existing.RouteProperties == nil {
 		return fmt.Errorf("retrieving existing %s: 'properties' was nil", *id)
 	}
 
-	var checkProtocols bool
+	// we need to lock the route for update because the custom domain
+	// association may also be trying to update the route as well...
+	locks.ByName(id.RouteName, cdnFrontDoorRouteResourceName)
+	defer locks.UnlockByName(id.RouteName, cdnFrontDoorRouteResourceName)
+
 	httpsRedirect := d.Get("https_redirect_enabled").(bool)
 	protocolsRaw := d.Get("supported_protocols").(*pluginsdk.Set).List()
+	customDomainsRaw := d.Get("cdn_frontdoor_custom_domain_ids").([]interface{})
+	originGroupRaw := d.Get("cdn_frontdoor_origin_group_id").(string)
+	ruleSetIdsRaw := d.Get("cdn_frontdoor_rule_set_ids").([]interface{})
+	linkToDefaultDomain := d.Get("link_to_default_domain").(bool)
+
+	// NOTE: If HTTPS Redirect is enabled the Supported Protocols must support both HTTP and HTTPS
+	// This configuration does not cause an error when provisioned, however the http requests that
+	// are supposed to be redirected to https remain http requests
+	if httpsRedirect {
+		if err := validate.SupportsBothHttpAndHttps(protocolsRaw, "https_redirect_enabled"); err != nil {
+			return err
+		}
+	}
 
-	props := azuresdkhacks.RouteUpdatePropertiesParameters{
-		CustomDomains: existing.RouteProperties.CustomDomains,
+	originGroup, err := parse.FrontDoorOriginGroupIDInsensitively(originGroupRaw)
+	if err != nil {
+		return err
+	}
+
+	normalizedCustomDomains, err := customDomainsInsensitively(customDomainsRaw)
```

Comment:
this is coming from the config/is validated, so should be being parsed case-sensitively

Reply:
Fixed.